src/translib/acl_app.go

```diff
@@ -313,6 +315,83 @@ func (app *AclApp) translateCRUCommon(d *db.DB, opcode int) ([]db.WatchKeys, err
 	return keys, err
 }
 
+func (app *AclApp) processAclGet(dbs [db.MaxDB]*db.DB) error {
+
+	var err error
+	var topmostPath bool = false
+
+	d := dbs[db.ConfigDB]
+	acl := app.getAppRootObject()
+
+	log.Infof("processAclGet--Path Received: %s", app.pathInfo.Template)
+	targetType := reflect.TypeOf(*app.ygotTarget)
+	if !util.IsValueScalar(reflect.ValueOf(*app.ygotTarget)) && util.IsValuePtr(reflect.ValueOf(*app.ygotTarget)) {
+		log.Infof("processAclGet: Target object is a <%s> of Type: %s", targetType.Kind().String(), targetType.Elem().Name())
+		if targetType.Elem().Name() == "OpenconfigAcl_Acl" {
+			topmostPath = true
+		}
+	}
+
+	targetUriPath, _ := getYangPathFromUri(app.pathInfo.Path)
+	if isSubtreeRequest(app.pathInfo.Template, "/openconfig-acl:acl/acl-sets") {
+		if isSubtreeRequest(app.pathInfo.Template, "/openconfig-acl:acl/acl-sets/acl-set{}{}") {
+			for aclSetKey, _ := range acl.AclSets.AclSet {
+				aclSet := acl.AclSets.AclSet[aclSetKey]
+				aclKey := getAclKeyStrFromOCKey(aclSetKey.Name, aclSetKey.Type)
+
+				if isSubtreeRequest(app.pathInfo.Template, "/openconfig-acl:acl/acl-sets/acl-set{}{}/acl-entries/acl-entry{}") {
+					// Subtree of one Rule
+					for seqId, _ := range aclSet.AclEntries.AclEntry {
+						entrySet := aclSet.AclEntries.AclEntry[seqId]
+						err = app.convertDBAclRulesToInternal(dbs , aclKey, int64(seqId), db.Key{})
+						ygot.BuildEmptyTree(entrySet)
+						app.convertInternalToOCAclRule(aclKey, aclSetKey.Type, int64(seqId), nil, entrySet)
+					}
+				} else {
+					err = app.convertDBAclToInternal(dbs, db.Key{Comp: []string{aclKey}})
+					if err != nil {
+						return err
+					}
+					ygot.BuildEmptyTree(aclSet)
+					app.convertInternalToOCAcl(aclKey, acl.AclSets, aclSet)
+				}
+			}
+
+		} else {
+			// Get all Acls and their rules
+			err = app.processCommonToplevelGetPath(dbs, acl, false)
+		}
+	} else if isSubtreeRequest(app.pathInfo.Template, "/openconfig-acl:acl/interfaces") {
+		if isSubtreeRequest(app.pathInfo.Template, "/openconfig-acl:acl/interfaces/interface{}") {
+			for intfId := range acl.Interfaces.Interface {
+				intfData := acl.Interfaces.Interface[intfId]
+				ygot.BuildEmptyTree(intfData)
+				if isSubtreeRequest(targetUriPath, "/openconfig-acl:acl/interfaces/interface/ingress-acl-sets") {
+					err = app.getAclBindingInfoForInterfaceData(d, intfData, intfId, "INGRESS")
```

Comment:
Since it is a loop, an error could get overwritten during the next iteration.
Let's at least log in case of error.

Reply:
will do.